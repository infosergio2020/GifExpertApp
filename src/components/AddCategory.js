import React,{ useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        setInputValue( e.target.value );
        console.log('handleInputChange llamado');
    }

    const handleSubmit = (e)=>{
        console.log('hadleSubmit',inputValue);
        e.preventDefault();
        
        if(inputValue.trim().length > 2 ){
            setCategories( cats => [inputValue,...cats]);
            setInputValue('');
        }else {
            // alert('Debe tener mas de 2 caracteres');
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input className="input input-background" 
                   type="text"
                   value={inputValue}
                   onChange={handleInputChange}>
            </input>
        </form>
    )
}

// con esto prevengo que alguien que reutilize mi funcion se olvide de mandar una funcion como parametro
AddCategory.propTypes = { 
    setCategories: PropTypes.func.isRequired,
}