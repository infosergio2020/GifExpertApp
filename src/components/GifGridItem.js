import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ( {title,url} ) => {
    return (
        <div className="card animate__animated animate__flipInX">
            <img src={url} alt={title}></img>
            <p> {title} </p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired, //indico que solo acepta este tipo
    url: PropTypes.string.isRequired
}


/*
    1. Enzyme
    2. Enzyme to Json
    3. Debe de mostrar el componente correctamente
        * shallow
        * wrapper
        * wrapper .toMatchSnapshot()
*/