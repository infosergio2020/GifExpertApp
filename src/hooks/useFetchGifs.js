import { useState,useEffect } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( category ) => {
    const key = 'dpoD26YubbJ6ixDgvXuW7Qc9cH7vVvxc';
    const value = 10;
    const [state, setState] = useState({
        data:[],
        loading: true
    });

    useEffect(()=>{
        getGifs(category,value,key)
            .then( imgs => {
                    setState({
                        data:imgs,
                        loading:false
                    });
            })
    }, [ category ]);

    return state;
}