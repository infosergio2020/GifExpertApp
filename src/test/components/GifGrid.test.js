import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid >', () => {
    const category = 'One Punch';
    test('debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } />)
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar items cuando se cargan imagenes', () => {
        
        
        const gifs = [
            {
                id:'ABC',
                url:'https://cualquiercosa.com',
                title: 'CualquierCosa'
            },
            {
                id:'123',
                url:'https://cualquiercosa.com',
                title: 'CualquierCosa'
            }
        ]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={ category } />)
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe( gifs.length );
    })
    
})
