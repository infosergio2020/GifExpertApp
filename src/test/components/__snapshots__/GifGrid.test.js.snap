// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid > debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="CualquierCosa"
      url="https://cualquiercosa.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="CualquierCosa"
      url="https://cualquiercosa.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid > debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
